core/options.c

```diff
@@ -618,57 +730,220 @@ check_param_bounds(uint *val, uint min, uint max, const char *name)
     return ret;
 }
 
-/* Print an option string from an option struct.
- * Xref case 7939, in DEBUG builds the ? : creates an unshared implicit local
- * that led to huge stack usage for update_dynamic_options() so we use
- * methods instead of macros for those.  While we could make these inline
- * (which DEBUG will ignore and in release will result in the same code as
- * MACRO version) these are way off the hot path and this way cuts the
- * size of the release build dll by ~7kb.
```

Comment:
This comment seems very relevant and the gist of it should be kept: there's a reason we're doing all these functions.